File: Codeforces/1334A.cpp
#include <bits/stdc++.h>
#define endl '\n'
using namespace std;

int main(){
    ios_base::sync_with_stdio(false);
    cin.tie(0);
    int t,n, antX, antY,x,y;
    bool ans;
    cin >> t;
    while(t--){
        cin >> n;
        cin >> antX >> antY;
        ans = true;
        if(antY > antX){
            ans = false;
        }
        for(int i = 0; i < n - 1; i++ ){
                cin >> x >> y;
                if(y > x){
                    ans = false;
                }
                else if(x < antX || y < antY){
                    ans = false;
                }
                else if(abs(y-antY) > abs(antX-x)){
                    ans = false;
                }
                antX = x;
                antY = y;
            }
        }
        if(ans){
            cout << "YES" << endl;
        }
        else{
            cout << "NO" << endl;
        }
    }
    return 0;

}

File: Codeforces/1334A.py
t = int(input())
for i in range(t):
    n = int(input())
    antX,antY = map(int,input().split())
    ans = True
    if antY > antX:
        ans = False
    for j in range(n-1):
        x,y = map(int,input().split())
        if y > x:
            ans = False
        elif x < antX or y < antY:
            ans = False
        elif abs(y-antY) > abs(antX-x):
            ans = False
        antY = y
        antX = x

    print('YES' if ans else 'NO')
 